plugin/src/main/java/org/opensearch/ml/cluster/MLSyncUpCron.java

```diff
@@ -91,6 +129,171 @@ public void run() {
+                        MLModel.MODEL_STATE_FIELD,
+                        MLModel.PLANNING_WORKER_NODE_COUNT_FIELD,
+                        MLModel.LAST_UPDATED_TIME_FIELD,
+                        MLModel.CURRENT_WORKER_NODE_COUNT_FIELD },
+                    null
+                );
+            searchRequest.source(sourceBuilder);
+            client.search(searchRequest, ActionListener.wrap(res -> {
+                SearchHit[] hits = res.getHits().getHits();
+                Map<String, MLModelState> newModelStates = new HashMap<>();
+                for (SearchHit hit : hits) {
+                    String modelId = hit.getId();
+                    Map<String, Object> sourceAsMap = hit.getSourceAsMap();
+                    MLModelState state = MLModelState.from((String) sourceAsMap.get(MLModel.MODEL_STATE_FIELD));
+                    Long lastUpdateTime = sourceAsMap.containsKey(MLModel.LAST_UPDATED_TIME_FIELD)
+                        ? (Long) sourceAsMap.get(MLModel.LAST_UPDATED_TIME_FIELD)
+                        : null;
+                    int planningWorkerNodeCount = sourceAsMap.containsKey(MLModel.PLANNING_WORKER_NODE_COUNT_FIELD)
+                        ? (int) sourceAsMap.get(MLModel.PLANNING_WORKER_NODE_COUNT_FIELD)
+                        : 0;
+                    int currentWorkerNodeCountInIndex = sourceAsMap.containsKey(MLModel.CURRENT_WORKER_NODE_COUNT_FIELD)
+                        ? (int) sourceAsMap.get(MLModel.CURRENT_WORKER_NODE_COUNT_FIELD)
+                        : 0;
+                    MLModelState mlModelState = getNewModelState(
+                        loadingModels,
+                        modelWorkerNodes,
+                        modelId,
+                        state,
+                        lastUpdateTime,
+                        planningWorkerNodeCount,
+                        currentWorkerNodeCountInIndex
+                    );
+                    if (mlModelState != null) {
+                        newModelStates.put(modelId, mlModelState);
+                    }
+                }
+                bulkUpdateModelState(modelWorkerNodes, newModelStates);
+            }, e -> {
+                updateModelStateSemaphore.release();
+                log.error("Failed to search models", e);
+            }));
+        } catch (Exception e) {
+            updateModelStateSemaphore.release();
+            log.error("Failed to refresh model state", e);
+        }
+    }
+
+    private MLModelState getNewModelState(
+        Map<String, Set<String>> loadingModels,
+        Map<String, Set<String>> modelWorkerNodes,
+
+        String modelId,
+        MLModelState state,
+        Long lastUpdateTime,
+        int planningWorkerNodeCount,
+        int currentWorkerNodeCountInIndex
+    ) {
+        Set<String> loadTaskNodes = loadingModels.get(modelId);
+        if (loadTaskNodes != null && loadTaskNodes.size() > 0 && state != MLModelState.LOADING) {
+            // no
```

Comment:
can we make a clear comment here?

Reply:
Seems forgot to include this change when commit. Will fix this part in next PR.